cmake/SpectreLoadMetadata.cmake

```diff
@@ -0,0 +1,57 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Read the following information from the `Metadata.yaml` file:
+# - SPECTRE_NAME: Name
+# - SPECTRE_VERSION: Version
+# - SPECTRE_DOI: Doi
+# - SPECTRE_ZENODO_ID: ZenodoId
+
+file(READ "${CMAKE_SOURCE_DIR}/Metadata.yaml" SPECTRE_METADATA_FILE)
+# Split lines into cmake list (see https://cmake.org/pipermail/cmake/2007-May/014222.html)
+string(REGEX REPLACE ";" "\\\\;" SPECTRE_METADATA_FILE "${SPECTRE_METADATA_FILE}")
+string(REGEX REPLACE "\n" ";" SPECTRE_METADATA_FILE "${SPECTRE_METADATA_FILE}")
+
+# Read data from the file
+set(SPECTRE_NAME "")
+set(SPECTRE_VERSION "")
```

Comment:
Should maybe also parse out the homepage for completeness? We might want to print that at startup of executables too. I'm not sure if we use the homepage anywhere in the docs, but conceivably that would also be easier to maintain if we have it in CMake

Reply:
And maybe also add a `GitRepo` that we parse out? I can see that being used in a few places (like in this PR :D )

Reply:
Added the `SPECTRE_HOMEPAGE` to cmake.

The metadata already has the `GitHub: sxs-collaboration/spectre` field. When parsing it out I couldn't get Doxygen to build a URL like `https://github.com/\spectregithub/xyz` though. The other place where we use the repo name is in the CI workflow files, which we don't CMake-configure, so can't insert the variable. So I'd just leave as-is. If other use-cases turn up we can always parse more metadata into cmake when needed.

Reply:
sounds good